src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.targets

```diff
@@ -138,10 +138,10 @@ Copyright (c) .NET Foundation. All rights reserved.
                  FormatArguments="$(SdkResolverGlobalJsonPath)" />
   </Target>
 
-  <Target Name="_WarnWhenUsingNET9AndVSPriorTo1712"
+  <Target Name="_WarnWhenUsingNET10AndVSPriorTo18"
         BeforeTargets="_CheckForInvalidConfigurationAndPlatform"
-        Condition="$([MSBuild]::VersionLessThan($(MSBuildVersion), '17.12.0')) and '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '9.0'))">
-    <NetSdkWarning ResourceName="Net9NotCompatibleWithDev1711"/>
+        Condition="$([MSBuild]::VersionLessThan($(MSBuildVersion), '18.0.0')) and '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '10.0'))">
```

Comment:
Is it intentional that we are checking `TargetFramework` here?

Issues happen whenever MSBuild 17 is used with .NET 10 SDK regardless of the TFM being targeted (unless we are talking about different issues here)

Reply:
According to @marcpopMSFT in #50748:

> Using the 10.0.1xx SDK to **target** net10.0 is not supported unless you're using VS 18.0.0.

> **Targeting** net10 will require VS 2026 as there are matching IDE side changes.

(emphasis mine)